src/iop/lens.cc

```diff
@@ -2749,8 +2747,8 @@ static void _distort_mask_md(struct dt_iop_module_t *self,
       const float dr =
         _interpolate_linear_spline(d->knots_dist, d->cor_rgb[1],
                                    d->nc, r*sqrtf(cx*cx + cy*cy));
-      const float xs = dr*cx + w2 - roi_in->x;
-      const float ys = dr*cy + h2 - roi_in->y;
+      const float xs = MIN(limw, MAX(0.0f, dr*cx + w2 - roi_in->x));
+      const float ys = MIN(limh, MAX(0.0f, dr*cy + h2 - roi_in->y));
```

Comment:
This is just CLAMP(), which would be clearer.

Reply:
Right.

Reply:
fixed, squashed and force-pushed